pipes/WDL/dx-launcher/demux_launcher.yml

```diff
@@ -0,0 +1,179 @@
+  execDepends:
+    - name: pigz
+    - name: libxml2-utils
+  interpreter: bash
+  code: |
+    #!/bin/bash
+
+    main() {
+      set -ex -o pipefail
+
+      # Download the RunInfo.xml file
+      runInfo_file_id=$(dx get_details "$upload_sentinel_record" | jq .runinfo_file_id -r)
+      dx cat $runInfo_file_id > RunInfo.xml
+
+      if [ -n "$api_token" ]; then
+        dx download -o api_token "$api_token"
+      fi
+
+      # Parse the lane count & run ID from RunInfo.xml file
+      lane_count=$(xmllint --xpath "string(//Run/FlowcellLayout/@LaneCount)" RunInfo.xml)
+      if [ -z "$lane_count" ]; then
+          dx-jobutil-report-error "Could not parse LaneCount from RunInfo.xml. Please check RunInfo.xml is properly formatted"
+      fi
+
+      surface_count=$(xmllint --xpath "string(//Run/FlowcellLayout/@SurfaceCount)" RunInfo.xml)
+      if [ -z "$surface_count" ]; then
+          dx-jobutil-report-error "Could not parse SurfaceCount from RunInfo.xml. Please check RunInfo.xml is properly formatted"
+      fi
+
+      swath_count=$(xmllint --xpath "string(//Run/FlowcellLayout/@SwathCount)" RunInfo.xml)
+      if [ -z "$swath_count" ]; then
+          dx-jobutil-report-error "Could not parse SwathCount from RunInfo.xml. Please check RunInfo.xml is properly formatted"
+      fi
+
+      tile_count=$(xmllint --xpath "string(//Run/FlowcellLayout/@TileCount)" RunInfo.xml)
+      if [ -z "$tile_count" ]; then
+          dx-jobutil-report-error "Could not parse TileCount from RunInfo.xml. Please check RunInfo.xml is properly formatted"
+      fi
+      
+      run_id=$(xmllint --xpath "string(//Run/@Id)" RunInfo.xml)
+      if [ -z "$run_id" ]; then
+          dx-jobutil-report-error "Could not parse Run@Id from RunInfo.xml. Please check RunInfo.xml is properly formatted"
+      fi
+
+      # total data size more roughly tracks total tile count
+      total_tile_count=$((lane_count*surface_count*swath_count*tile_count))
+
+      demux_instance_type="mem1_ssd1_x4"
+      if [ "$total_tile_count" -le 50 ]; then 
+          tar_consolidation_instance_size="mem1_ssd1_x4"
+          demux_instance_type="mem1_ssd1_x4"
+          min_base_quality=25
+          echo "Detected $total_tile_count tiles, interpreting as MiSeq run, executing on a $demux_instance_type machine."
+      elif [ "$total_tile_count" -le 150 ]; then
+          tar_consolidation_instance_size="mem1_ssd2_x4"
+          demux_instance_type="mem1_ssd2_x4"
+          min_base_quality=25
+          echo "Detected $total_tile_count tiles, interpreting as HiSeq2k run, executing on a $demux_instance_type machine."
+      elif [ "$total_tile_count" -le 896 ]; then
+          tar_consolidation_instance_size="mem1_hdd2_x32"
+          demux_instance_type="mem1_hdd2_x32"
+          min_base_quality=25
+          echo "Detected $total_tile_count tiles, interpreting as HiSeq4k run, executing on a $demux_instance_type machine."
+      elif [ "$total_tile_count" -le 1408 ]; then
```

Comment:
I had noticed a while ago that I've had difficulty getting `hdd2` AWS/DX instances to run. @mlin can you confirm if these are being phased out? Should we stick to `ssd2` instances instead?